homeassistant/components/switchbot/__init__.py

```diff
@@ -78,21 +70,14 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         raise ConfigEntryNotReady(
             f"Could not find Switchbot {sensor_type} with address {address}"
         )
-
-    if COMMON_OPTIONS not in domain_data:
```

Comment:
We need to update the config entry update listener below too.

Reply:
I had to wrap it in the class to hold a strong reference to it since the weakref kept getting destroyed